### PR TITLE
Added AzCopy 10.14.0 Changelogs

### PR DESCRIPTION
**Pull request by mohsha-msft**

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,6 +1,32 @@
 
 # Change Log
 
+## Version 10.14.0
+
+### New features
+1. Feature to [permanently delete](https://docs.microsoft.com/en-us/rest/api/storageservices/delete-blob#remarks) soft-deleted
+   snapshots/versions of the blobs has been added (preview). `--permanent-delete=none/snapshots/version/snapshotsandversions`.
+2. Feature to preserve properties and ACLs when copying to Azure file share root directory.
+3. Pin all APIs to use the default service version `2020-04-08` and let users decide the service version via
```
Comment (zezha-msft) [inline ChangeLog.md:10]: clarify that previously not all APIs respected the AZCOPY_DEFAULT_SERVICE_API_VERSION.

File: ChangeLog.md [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+   `AZCOPY_DEFAULT_SERVICE_API_VERSION` environment variable.
+
+### Bug fixes
+1. Fixed issue in which AzCopy failed to copy to classic blob container with `preserve blob access tier`.
+2. Fixed [issue 1630](https://github.com/Azure/azure-storage-azcopy/issues/1630) : AzCopy created extra empty
+   directories at destination while performing S2S transfer from one ADLS Gen2 account to another ADLS Gen2 account.
+3. Changed the way we were using to obtain and set ACLs to ensure accuracy.
```
Comment (zezha-msft) [inline ChangeLog.md:17]: Style: avoid "we", use "AzCopy" instead

File: ChangeLog.md [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+5. Report error when client provided key(CPK) encryption is applied to DFS endpoint.
+6. Fixed [issue 1596](https://github.com/Azure/azure-storage-azcopy/issues/1596) : AzCopy failed to transfer files 
+   (with '/.' in their path) from AWS S3 to Azure blob storage.
+7. Fixed [issue 1474](https://github.com/Azure/azure-storage-azcopy/issues/1474) : AzCopy panicked when trying to re-create an already open plan file.
+8. Improved handling of Auth error against single file.
+9. Fixed [issue 1640](https://github.com/Azure/azure-storage-azcopy/issues/1640) : Recursive copy from GCS bucket to Azure container failed 
+   with `FileIgnored` error when using `--exclude-path`.
+10. Fixed [issue 1655](https://github.com/Azure/azure-storage-azcopy/issues/1655) : AzCopy panicked when using `--include-before` flag.
+11. Fixed [issue 1609](https://github.com/Azure/azure-storage-azcopy/issues/1609) : `blockid` converted to lower case in AzCopy logs.
+12. Fixed [issue 1643](https://github.com/Azure/azure-storage-azcopy/issues/1643), [issue 1661](https://github.com/Azure/azure-storage-azcopy/issues/1661) : Updated Golang version to `1.17.7` to fix security vulnerabilities in Golang packages.
```
Comment (zezha-msft) [inline ChangeLog.md:28]: nice, I like the issue links

**Review by zezha-msft — APPROVED** [review]
Lgtm.

Please check out the test failure, is it a flaky test?

**Comment by zezha-msft** [comment]
Please also update the version number in common/version.go.

**Review by nakulkar-msft — APPROVED** [review]

**Pull request merged**